packages/@azure/servicebus/data-plane/lib/util/utils.ts

```diff
@@ -174,3 +175,15 @@ export function toBuffer(input: any): Buffer {
   log.utils("[utils.toBuffer] The converted buffer is: %O.", result);
   return result;
 }
+
+/**
+ * Throws InvalidOperationError if the current AMQP connection is closed.
+ * @param context The ConnectionContext associated with the current AMQP connection.
+ */
+export function throwErrorIfConnectionClosed(context: ConnectionContext): void {
```

Comment:
Would it be possible to free up memory references by setting client context to undefined when close() is invoked?

Reply:
Not sure where you'd set this, but generally this isn't worth doing in JS. In practice, when users are done with the client, they'll drop the reference to it (e.g. the reference will go out of scope), and then the entire memory graph attached to the client gets collected (unless bits of it are held by e.g. long-lived callbacks we haven't de-registered).

Reply:
While we dont need to worry about this when closing the namespace, we should indeed look into what happens when the individual queue/topic/subscription clients are closed. Since `context` will hold on to references to these, they will lie around for the whole lifetime of the namespace